app/src/main/java/org/mozilla/fenix/utils/ClearableEditText.kt

```diff
@@ -24,26 +24,23 @@ class ClearableEditText @JvmOverloads constructor(
     AppCompatEditText(context, attrs, defStyleAttr) {
 
     @SuppressLint("ClickableViewAccessibility")
-    override fun onTouchEvent(event: MotionEvent?): Boolean {
-        if (length() != 0 && event?.action == MotionEvent.ACTION_UP &&
-            event.rawX >= (this@ClearableEditText.right - this@ClearableEditText.compoundPaddingRight)
-        ) {
-            this@ClearableEditText.setText("")
+    override fun onTouchEvent(event: MotionEvent): Boolean {
+        if (length() != 0 && event.action == ACTION_UP && event.rawX >= (right - compoundPaddingRight)) {
+            setText("")
             return true
         }
         return super.onTouchEvent(event)
     }
 
     override fun onTextChanged(text: CharSequence?, start: Int, lengthBefore: Int, lengthAfter: Int) {
         super.onTextChanged(text, start, lengthBefore, lengthAfter)
-        if (lengthAfter != 0 && error == null) {
-            setCompoundDrawablesWithIntrinsicBounds(0, 0, R.drawable.ic_clear, 0)
-            for (drawable: Drawable in compoundDrawables.filterNotNull()) {
-                val color = ThemeManager.resolveAttribute(R.attr.primaryText, context!!)
-                drawable.colorFilter = PorterDuffColorFilter(getColor(context, color), SRC_IN)
+        val drawable = if (lengthAfter != 0 && error == null) {
+            context.getDrawable(R.drawable.ic_clear)?.apply {
+                colorFilter = PorterDuffColorFilter(R.attr.primaryText.getColorFromAttr(context), SRC_IN)
```

Comment:
This took me a second to work out where the loop had gone.

The `lengthAfter != 0` is a little confusing. Could put a comment on the intent of this?

Reply:
Added comments to the class. The loop actually just changes the end drawable, so I removed it.